### ISSUE TITLE
Make the plugin fully federated

### ISSUE THREAD

**Issue by bartekpacia** [MaikuB/flutter_local_notifications#1662]
I see that there is `flutter_local_notifications_platform_interface` and  `flutter_local_notifications_linux` packages. But implementation for `android`, `ios`, and `macos` are still done "the old way". Why is it like that @MaikuB?

**Comment by MaikuB** [comment]
Unlikely most of the plugins in the ecoystem, this one has a like of platform-specific classes. There was no mandate to split each platform implementation into its own package either and it's only been a relative recent thing that the first-party plugins have done this. This is why the plugin is the way it is. Perhaps it could be split now and others in the community like yourself welcome to contribute to do so.

I'm not sure what you were trying to link to as it just goes the Java file where most of the Android logic resides. I'm assuming you were trying to point to the `registerWith` method. If so, you should read https://docs.flutter.dev/development/platform-integration/android/plugin-api-migration. You'll find other plugins still support it following the guidance there

**Comment by MaikuB** [comment]
Forgot to add that if the plugin was going to be split per platform then this could cause this more difficulties for iOS and macOS as pllatform-specific classes/APIs should really reside with the platform implementations. In the 10.0 prerelease, you'll find that I had been refactoring the iOS and macOS classes that they're now prefixed with "Darwin"

**Comment by bartekpacia** [comment]
> I'm not sure what you were trying to link to as it just goes the Java file where most of the Android logic resides. I'm assuming you were trying to point to the registerWith method. If so, you should read https://docs.flutter.dev/development/platform-integration/android/plugin-api-migration. You'll find other plugins still support it following the guidance there.

Yes, that's exactly what I was asking about. Thanks for answering!

>  In the 10.0 prerelease, you'll find that I had been refactoring the iOS and macOS classes that they're now prefixed with "Darwin"

Nice :) If you don't have anything against it, I'd be interested in making this plugin fully federated once 10.0 is released (I didn't look into it but I suspect there'll be some breaking changes).

**Comment by MaikuB** [comment]
I don't have anything against it :)

**Comment by MaikuB** [comment]
This is now released though I had to publish version 11 to account for a bump in the `timezone` package. Shall I assign this issue to you then?

**Comment by bartekpacia** [comment]
Congrats on the new release!

And regarding assigning me – sure, please assign me :)

**Comment by okbye123** [comment]
This will help https://www.youtube.com/watch?v=aQTvI8L-mZs

**Comment by MaikuB** [comment]
Hey @bartekpacia wondering if you by chance made progress on this? I'm assuming not as you're probably busy with other things like patrol :)

**Comment by bartekpacia** [comment]
Hi, you're right, I didn't do anything. You can unassign me, maybe someone else will pick it up :)

**Comment by bartekpacia** [comment]
Oh, I could do it myself.
